os/kernel/log_dump/log_dump.c

```diff
@@ -176,6 +177,14 @@ int log_dump_compress_lastblock(void)
 {
 	read_node = (struct log_dump_chunk_s *)sq_peek(&log_dump_chunks);	/* reset the read node to head */
 
+	if (read_last_n_nodes > 0) {
+		int nodes = read_last_n_nodes;
```

Comment:
Why do we need to make a separate variable called nodes?

Reply:
Done, used the same variable.